lib/Twig/Loader/Chain.php

```diff
@@ -66,6 +66,35 @@ public function getSource($name)
     /**
      * {@inheritdoc}
      */
+    public function getSourceContext($name)
+    {
+        $exceptions = array();
+        foreach ($this->loaders as $loader) {
+            if (!$loader instanceof Twig_SourceContextLoaderInterface) {
+                continue;
```

Comment:
this is broken in case this loader was the one loading the template, but a next loader could load a template with the same name